jaxrs-api/src/main/java/jakarta/ws/rs/HttpVersionNotSupportedException.java

```diff
@@ -0,0 +1,105 @@
+/********************************************************************************
+ * Copyright (c) 2019 Jeyvison Nascimento
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ ********************************************************************************/
+
+package jakarta.ws.rs;
+
+import jakarta.ws.rs.core.Response;
+
+/**
+ * A runtime exception indicating that The server does not support, or refuses to support,
+ * the HTTP protocol version that was used in the request message.
+ *
+ * @author Jeyvison Nascimento (jeynoronha@gmail.com)
+ * @since 2.2
+ */
+public class HttpVersionNotSupportedException extends ServerErrorException {
+
+    private static final long serialVersionUID = -2162737983746617138L;
+
+    /**
+     * Construct a new bad client request exception.
```

Comment:
I'm not sure why this is described as "bad client request exception". It's very generic, whereas 505 is more specific. Same throughout the definition of HttpVersionNotSupportedException.